source/src/core/conformation/util.cc

```diff
@@ -319,20 +320,36 @@ idealize_position(
 
 	if ( rsd.is_polymer() ) {
 		// add polymer nbrs?
-		if ( seqpos > 1 && !rsd.is_lower_terminus() && !conformation.fold_tree().is_cutpoint( seqpos-1 ) ) {
-			lower_connect = true;
-			ResidueOP prev_rsd( ResidueFactory::create_residue( conformation.residue( seqpos-1 ).type() ) );
-			idl.prepend_polymer_residue_before_seqpos( *prev_rsd, 1, true );
-			idl_pos = 2;
-		}
+		if ( rsd.is_carbohydrate() ) {
+			if ( seqpos > 1 && !rsd.is_lower_terminus() ) {
+				lower_connect = true;
+				core::Size parent_res_seqpos( conformation::carbohydrates::find_seqpos_of_saccharides_parent_residue( rsd ) );
+				ResidueOP prev_rsd( ResidueFactory::create_residue( conformation.residue( parent_res_seqpos ).type() ) );
+				idl.prepend_polymer_residue_before_seqpos( *prev_rsd, 1, true );
+				idl_pos = 2;
+			}
+
+			if ( seqpos < conformation.size() && !rsd.is_upper_terminus() && !conformation.fold_tree().is_cutpoint( seqpos ) ) {
+				upper_connect = true;
+				core::Size child_res_seqpos( conformation::carbohydrates::find_seqpos_of_saccharides_mainchain_child( rsd ) );
+				ResidueOP next_rsd( ResidueFactory::create_residue( conformation.residue( child_res_seqpos ).type() ) );
+				idl.append_polymer_residue_after_seqpos( *next_rsd, idl_pos, true );
+			}
+		} else {
+			if ( seqpos > 1 && !rsd.is_lower_terminus() && !conformation.fold_tree().is_cutpoint( seqpos-1 ) ) {
+				lower_connect = true;
+				ResidueOP prev_rsd( ResidueFactory::create_residue( conformation.residue( seqpos-1 ).type() ) );
```

Comment:
The assumptions that a polymer is connected to `seqpos-1` & `seqpos+1` isn't necessarily valid, even in the case of proteins. -- But that's pre-existing code, so it shouldn't necessarily limit this PR. (Though if there was a generalization of UPPER/LOWER determination which works with carbohydrates, that might allow us to clean this up.)